### PR TITLE
fixup! [IMP] point_of_sale: Store applied pricelist item on Orderline

### PR DESCRIPTION
**Pull request by QuocDuong1306**

### CODE REVIEW

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -1567,7 +1567,7 @@ class Product extends PosModel {
     // product.pricelist.item records are loaded with a search_read
     // and were automatically sorted based on their _order by the
     // ORM. After that they are added in this order to the pricelists.
-    get_price(pricelist, quantity, price_extra){
+    get_price(pricelist, quantity, price_extra, get_pricelist_rule=false){
```
Comment (nilshamerlinck) [inline addons/point_of_sale/static/src/js/models.js]:
```suggestion
    get_price(pricelist, quantity, price_extra, get_pricelist_item=false){
```

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -1637,7 +1637,7 @@ class Product extends PosModel {
         // being used further. Note that this cannot happen here,
         // because it would cause inconsistencies with the backend for
         // pricelist that have base == 'pricelist'.
-        return price;
+        return get_pricelist_rule ? {price: price, pricelist_item: applied_item}: price;
```
Comment (nilshamerlinck) [inline addons/point_of_sale/static/src/js/models.js]:
```suggestion
        return get_pricelist_item ? {price: price, pricelist_item: applied_item}: price;
```

File: addons/point_of_sale/static/src/js/models.js [hunk 3]
```diff
@@ -2079,6 +2086,10 @@ class Orderline extends PosModel {
 
         return wrapped;
     }
+    set_pricelist_item(rule){
```
Comment (nilshamerlinck) [inline addons/point_of_sale/static/src/js/models.js]:
```suggestion
    set_pricelist_item(item){
```

File: addons/point_of_sale/static/src/js/models.js [hunk 4]
```diff
@@ -2079,6 +2086,10 @@ class Orderline extends PosModel {
...
+        this.order.assert_editable();
+        this.pricelist_item = rule;
```
Comment (nilshamerlinck) [inline addons/point_of_sale/static/src/js/models.js]:
```suggestion
        this.pricelist_item = item;
```

File: addons/point_of_sale/static/src/js/models.js [hunk 5]
```diff
@@ -2016,6 +2022,7 @@ class Orderline extends PosModel {
             refunded_orderline_id: this.refunded_orderline_id,
             price_manually_set: this.price_manually_set,
             price_automatically_set: this.price_automatically_set,
+            pricelist_item_id: this.get_pricelist_item() ? this.get_pricelist_item().id : false,
```
Comment (nilshamerlinck) [inline addons/point_of_sale/static/src/js/models.js]:
```suggestion
            pricelist_item_id: this.pricelist_item?.id ?? false,
```

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]
nitpicks

**Comment by QuocDuong1306** [comment]
Thank Nils, The PR is updated

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]
I was thinking in the shower, maybe worth showing the new line field in a backend view somewhere?

**Comment by QuocDuong1306** [comment]
Hi nils, as I saw in sale.order: https://github.com/odoo/odoo/blob/4c91a4ba280e4709d391419cc9f6b39432bdbe5f/addons/sale/models/sale_order_line.py#L135

`pricelist_item_id` was just used to cache for calculating, and was not shown in the view: https://github.com/odoo/odoo/blob/16.0/addons/sale/views/sale_order_views.xml

**Comment by QuocDuong1306** [comment]
Do you want me to do it differently from sale.order on pos.order?

If yes, it can be shown right here: https://github.com/odoo/odoo/blob/4c91a4ba280e4709d391419cc9f6b39432bdbe5f/addons/point_of_sale/views/pos_order_view.xml#L80
with: `<field name="pricelist_item_id" optional="hide" />`

**Review by nilshamerlinck — APPROVED** [review]
thanks for the research, good as is then

**Comment by nilshamerlinck** [comment]
ah but just to be sure: these new changes will be for master branch only, not for alloboissons

**Pull request closed without merging**